packages/solana/src/client.ts

```diff
@@ -229,6 +231,19 @@ export class Web3Modal extends Web3ModalScaffold {
       }
     })
 
+    EventsController.subscribe(state => {
+      if (state.data.event === 'SELECT_WALLET' && state.data.properties?.name === 'Phantom') {
+        const isMobile = CoreHelperUtil.isMobile()
+        if (isMobile && typeof window !== 'undefined' && !window.phantom) {
```

Comment:
@phoenixVS Could we use `CoreHelperUtil.isClient()` instead of `typeof window !== 'undefined'` sorry for late notice I forgot about it.